src/NHibernate/Async/Loader/Custom/CustomLoader.cs

```diff
@@ -52,6 +54,12 @@ public Task<IList> ListAsync(ISessionImplementor session, QueryParameters queryP
 			return rowProcessor.BuildResultRowAsync(row, rs, resultTransformer != null, session, cancellationToken);
 		}
 
+		protected override async Task PutResultInQueryCacheAsync(ISessionImplementor session, QueryParameters queryParameters, IType[] resultTypes,
+									   IQueryCache queryCache, QueryKey key, IList result, CancellationToken cancellationToken)
+		{
+			await base.PutResultInQueryCacheAsync(session, queryParameters, resultTypes, queryCache, key, result, cancellationToken);
```

Comment:
The wrong resultTypes are used.

Reply:
I thought I should use in parameter from PutResultInQueryCacheAsync method? Can u provide me with detail from where to take it. Thx

Reply:
You should do exactly what you've done in the non-async version. Ignore the parameter and send in this.resultTypes instead.

Reply:
In fact Async files should not be manually modified, just run the async generator (see contributing guidelines).

If there is a trouble with what it generates, then discuss about how to handle that.